rebac-admin-backend/v1/identities.go

```diff
@@ -0,0 +1,176 @@
+	_, err := h.Identities.DeleteIdentity(id)
+	if err != nil {
+		response := h.IdentitiesErrorMapper.MapError(err)
+		writeResponse(w, response.Status, response)
+		return
+	}
+
+	w.WriteHeader(200)
+}
+
+// GetIdentitiesItem Get a single identity.
+// (GET /identities/{id})
+func (h handler) GetIdentitiesItem(w http.ResponseWriter, req *http.Request, id string) {
+	identity, err := h.Identities.GetIdentity(id)
+	if err != nil {
+		response := h.IdentitiesErrorMapper.MapError(err)
+		writeResponse(w, response.Status, response)
+		return
+	}
+
+	writeResponse(w, 200, identity)
+}
+
+// PutIdentitiesItem Update an identity.
+// (PUT /identities/{id})
+func (h handler) PutIdentitiesItem(w http.ResponseWriter, req *http.Request, id string) {
+	identity := new(r.Identity)
+	defer req.Body.Close()
+
+	err := json.NewDecoder(req.Body).Decode(identity)
+	if err != nil {
+		writeErrorResponse(w, err)
+		return
+	}
+
+	if id != *identity.Id {
+		writeErrorResponse(w, NewValidationError("Identity ID from path does not match the Identity object"))
+		return
+	}
+
+	identity, err = h.Identities.UpdateIdentity(identity)
+	if err != nil {
+		response := h.IdentitiesErrorMapper.MapError(err)
+		writeResponse(w, response.Status, response)
+		return
+	}
+
+	writeResponse(w, 200, identity)
+}
+
+// GetIdentitiesItemEntitlements List entitlements the identity has.
+// (GET /identities/{id}/entitlements)
+func (h handler) GetIdentitiesItemEntitlements(w http.ResponseWriter, req *http.Request, id string, params r.GetIdentitiesItemEntitlementsParams) {
+	entitlements, err := h.Identities.GetIdentityEntitlements(id, &params)
+	if err != nil {
+		response := h.IdentitiesErrorMapper.MapError(err)
+		writeResponse(w, response.Status, response)
+		return
+	}
+
+	response := r.GetIdentityEntitlementsResponse{
+		Data:   entitlements.Data,
+		Status: 200,
+	}
+
+	writeResponse(w, 200, response)
+}
+
+// PatchIdentitiesItemEntitlements Add or remove entitlement to/from an identity.
+// (PATCH /identities/{id}/entitlements)
+func (h handler) PatchIdentitiesItemEntitlements(w http.ResponseWriter, req *http.Request, id string) {
+	w.WriteHeader(http.StatusNotImplemented)
+}
+
+// GetIdentitiesItemGroups List groups the identity is a member of.
+// (GET /identities/{id}/groups)
+func (h handler) GetIdentitiesItemGroups(w http.ResponseWriter, req *http.Request, id string, params r.GetIdentitiesItemGroupsParams) {
+	groups, err := h.Identities.GetIdentityGroups(id, &params)
+	if err != nil {
+		response := h.IdentitiesErrorMapper.MapError(err)
+		writeResponse(w, response.Status, response)
+		return
+	}
+
+	response := r.GetIdentityGroupsResponse{
+		Data:   groups.Data,
+		Status: 200,
+	}
+
+	writeResponse(w, 200, response)
+}
+
+// PatchIdentitiesItemGroups Add or remove the identity to/from a group.
+// (PATCH /identities/{id}/groups)
+func (h handler) PatchIdentitiesItemGroups(w http.ResponseWriter, req *http.Request, id string) {
```

Comment:
perhaps we could name this method something like `UpdateGroups`?

Reply:
Unfortunately the name is the one autogenerated by the tool based on the API spec, so renaming methods is not possible unless modifying the OpenAPI spec with "extension properties" that relate to the specific tool we used.
I'm updateing the godocs but method names will unfortunately stay the same

Reply:
@alesstimec Yeah, the methods on this `handler` struct are internal. The product services only deal with the abstractions defined in the `interfaces` package.